test/RequestStreamTest.cpp

```diff
@@ -32,9 +32,8 @@ class TestHandlerSync : public rsocket::RSocketResponder {
 };
 
 TEST(RequestStreamTest, HelloSync) {
-  auto port = randPort();
-  auto server = makeServer(port, std::make_shared<TestHandlerSync>());
-  auto client = makeClient(port);
+  auto server = makeServer(std::make_shared<TestHandlerSync>());
+  auto client = makeClient(*server->listeningPort());
```

Comment:
nit: I did not like the idea of having no listening port (optional), null return value. So we force the caller to always check if the return value is null, then the server might not be running. So even though the caller might be sure, they should always write if/else code. 
So we force this method to give information about if the server is really running or not.